Gems/GradientSignal/Code/Source/Components/ImageGradientComponent.cpp

```diff
@@ -226,8 +236,19 @@ namespace GradientSignal
         GradientRequestBus::Handler::BusConnect(GetEntityId());
         AZ::Data::AssetBus::Handler::BusConnect(m_configuration.m_imageAsset.GetId());
 
+        // If the image asset is already ready (e.g. constructed in a unit test),
+        // then go ahead and retrieve the image data now
         AZStd::unique_lock<decltype(m_imageMutex)> imageLock(m_imageMutex);
-        m_configuration.m_imageAsset.QueueLoad();
+        if (m_configuration.m_imageAsset.IsReady())
+        {
+            GetSubImageData();
+        }
+        // Otherwise for normal use-case, we queue the asset to be loaded now
+        else
+        {
+            m_imageData = AZStd::span<const uint8_t>();
+            m_configuration.m_imageAsset.QueueLoad();
+        }
```

Comment:
Is this change needed? If the asset is already ready, the connection to the AssetBus directly above should immediately call OnAssetReady(). You might just need to move the m_imageData & QueueLoad() calls to happen _before_ the AssetBus connect instead of after.

Reply:
Oh nice, thanks yes that does work! Updating now :)